source/code/plugins/oms_configuration.rb

```diff
@@ -135,7 +189,22 @@ def load_configuration(conf_path, cert_path, key_path)
 
         File.open(conf_path).each_line do |line|
           if line =~ /AZURE_RESOURCE_ID/
-            @@AzureResourceId = line.sub("AZURE_RESOURCE_ID=","").strip
+            azure_resource_id = get_azure_resid_from_imds()
+
+            retries = 0
+            until !azure_resource_id.nil? do
+              azure_resource_id = get_azure_resid_from_imds()
+              retries += 1
+              break if retries > 2
+              sleep 2
```

Comment:
Consider exponential backoff

Reply:
That would create delay in fluentd. Anyway, I'm planning on moving this logic to the thread function.